### PR TITLE
Ufcmap

### PR DESCRIPTION
**Pull request by king-mahdi**
UFCmap is added.

### CODE REVIEW

**Comment by mstrout** [comment]
It looks like you edited the existing UFCallMapAndBounds.  Instead create a UFCallMap and .cc and just put the functionality from UFCallMapAndBounds that we need.  Our goal will be to remove UFCallMapAndBounds once we move over to using your UFCallMap implementation.

File: src/set_relation/UFCallMap.cc [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/*!
+ * \file UFCallMap.cc
+ *
+ * \brief Interface of UFCallMap class
+ *
+ * \date Started: 2016-02-02
+ *
+ * \author Michelle Strout
```
Comment (mstrout) [inline src/set_relation/UFCallMap.cc]: Mahdi, put yourself in as an author as well.

File: src/set_relation/UFCallMap.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*!
+ * \file UFCallMap.h
+ *
+ * \brief Interface of UFCallMap class
+ *
+ * \date Started: 2016-02-02
+ *
+ * \author Michelle Strout
```
Comment (mstrout) [inline src/set_relation/UFCallMap.h]: Here as well.

**Comment by mstrout** [comment]
There needs to be some tests that check both directions of the mapping.  From a UFCallTerm to a string and vice versa.

**Comment by king-mahdi** [comment]
I applied the changes. Tests for checking both directions of the map was in
UFCallMap_test, but it was a bit hidden. I added some comments to make it
clear.

**Comment by mstrout** [comment]
Almost there.  Just a couple more things:
- Create an assignment operator for the class.  See examples in existing classes.  If have a copy constructor then should have an assignment operator.
- Throughout the rest of the library, we are passing in pointers to things like UFCallTerms instead of references.  Please change that and in the header file documentation for each method indicate that the map does not own the object pointed at.

**Comment by king-mahdi** [comment]
I applied the changes.

**Pull request merged**